### ISSUE TITLE
Spectral Covariance

### ISSUE THREAD

**Issue by iancze** [MPoL-dev/MPoL#18]
**Is your feature request related to a problem or opportunity? Please describe.**
Data from spectral correlator systems frequently (always?) has non-zero spectral covariance. This covariance should be propagated to the loss functions used to assess model fit, especially for ungridded datasets (#17). 

**Describe the solution you'd like**
After implementing ungridded datasets (#17) (as well as some time/baseline tracking), provide a covariance matrix across channels that reflects the windowing function used in the data acquisition. See TMS v3 Section 8.8.2 and [Loomis+18 Appendix C](https://ui.adsabs.harvard.edu/abs/2018AJ....155..182L/abstract) for more details.

**Describe alternatives you've considered**
For lower S/N datasets (and continuum datasets), ignoring this effect and using the GriddedDataset is adequate. But for high S/N applications, it'd be nice to get as much information out of the data as possible.

**Comment by iancze** [comment]
The following is a proposed first-order solution to the spectral response issue.

**Assumptions**
* All visibilities are sampled at the same spectral resolution, and that their Doppler corrections have been applied and interpolated to a set of common bins (using `mstransform`). 
* The interpolation procedure has a negligible effect on the spectral response relative to the Hanning window itself.

**Procedure**
The procedure works similar to a simple RML loop, with BaseCube configured to have the same channel spacing as the data. Then each forward pass calculates BaseCube -> ImageCube -> FourierCube
1. After the FourierCube step, a convolution with a Hann window is applied in the frequency domain to the model
2. The model visibilities are selected relative to the gridded dataset, yet retain their channelization (edge case where visibilities left one cell/relative to the other)
3. The negative likelihood loss is calculated using a covariance matrix accounting for the Hann convolution

**Concerns/doubts**
* Will there be significant edge cases where the visibility mask for GriddedDataset is different in each channel?
* Are there many use cases where would want to account for the spectral covariance but not for the interpolation response/Doppler setting?

**Comment by iancze** [comment]
**Doppler setting**
Where is the (thermal) "noise" introduced into the process of generating the visibility, and how does this affect our ability to interpolate/shift spectral data?

Would it be possible to do a Fourier Transform, phase shift, and then process all visibilities to a common channel center?

**Comment by iancze** [comment]
Given the various forms of uncertainty related to the calculation, one potential route is to parameterize the spectral covariance with a Gaussian process kernel + nugget term, and infer the hyperparameters of the kernel self-consistently with the data.
